src/settings/appsettings.h

```diff
@@ -325,6 +325,22 @@ class AppSettings : public QObject
     void setCopyTranslationShortcut(const QKeySequence &shortcut);
     static QKeySequence defaultCopyTranslationShortcut();
 
+    QKeySequence clearShortcut() const;
+    void setClearShortcut(const QKeySequence &shortcut);
+    static QKeySequence defaultClearShortcut();
+
+    QKeySequence switchSourceShortcut() const;
+    void setSwitchSourceShortcut(const QKeySequence &shortcut);
+    static QKeySequence defaultSwitchSourceShortcut();
+
+    QKeySequence switchTranslationShortcut() const;
+    void setSwitchTranslationShortcut(const QKeySequence &shortcut);
+    static QKeySequence defaultSwitchTranslationShortcut();
+
+    QKeySequence switchEngineShortcut() const;
+    void setSwitchEngineShortcut(const QKeySequence &shortcut);
+    static QKeySequence defaultSwitchEngineShortcut();
```

Comment:
You need to add definitions of these function in `appsettings.cpp`.

Reply:
I'm afraid i don't really know how this would be done

Reply:
Just look into similar functions and copy them with minor edits :)

Reply:
I would suggest to start with `clear` button. Just only this one in this PR to learn how it's usually done.